dvc/api/artifacts.py

```diff
@@ -36,12 +36,16 @@ def artifacts_show(
     if version and stage:
         raise ValueError("Artifact version and stage are mutually exclusive.")
 
+    from dvc.repo.artifacts import Artifacts
+
     repo_kwargs: dict[str, Any] = {
         "subrepos": True,
         "uninitialized": True,
     }
+
+    dirname, _ = Artifacts.parse_path(name)
     with Repo.open(repo, **repo_kwargs) as _repo:
         rev = _repo.artifacts.get_rev(name, version=version, stage=stage)
-        with _repo.switch(rev):
-            path = _repo.artifacts.get_path(name)
-        return {"rev": rev, "path": path}
+        with _repo.switch(rev), Repo(dirname, fs=_repo.fs, scm=_repo.scm) as r:
```

Comment:
How does this behave in the case of a normal/non-monorepo with `subdir/dvc.yaml:myartifact`?

Reply:
In case of `subdir/dvc.yaml`, `parse_path` will return `subdir` as a dirname.
`Repo("subdir")` will try to find the dvc repo that `subdir` falls in, by traversing up the path and looking for `.dvc` directory.

In "normal/non-monorepo" case, it'll _discover_ the main repo. We are opening the same repo twice, but there are no additional costs to it and seems to work fine without issues.